src/runtime_src/driver/include/xclbin.h

```diff
@@ -247,7 +249,14 @@ extern "C" {
     /* IPs on AXI lite - their types, names, and base addresses.*/
     struct ip_data {
         uint32_t m_type; //map to IP_TYPE enum
-        uint32_t properties; //32 bits to indicate ip specific property. eg if m_type == IP_KERNEL then bit 0 is for interrupt.
+        union {
+            uint32_t properties; //32 bits to indicate ip specific property. eg if m_type == IP_KERNEL then bit 0 is for interrupt.
+            struct {     // Used by IP_MEM_* types
+               uint32_t m_index;
```

Comment:
do you mean uint16_t here?

Reply:
Great catch...yes it needs to be uint16_t.